staging/maven-plugin/src/main/java/org/sonatype/nexus/maven/staging/workflow/CloseStageRepositoryMojo.java

```diff
@@ -42,31 +46,46 @@
   public void doExecute(final StagingWorkflowV2Service stagingWorkflow)
       throws MojoExecutionException, MojoFailureException
   {
-    final String[] stagingRepositoryIds = getStagingRepositoryIds();
-    try {
-      getLog().info("Closing staging repository with IDs=" + Arrays.toString(getStagingRepositoryIds()));
-      stagingWorkflow.finishStagingRepositories(getDescriptionWithDefaultsForAction(StagingAction.FINISH),
-          stagingRepositoryIds);
-      getLog().info("Closed");
-    }
-    catch (StagingRuleFailuresException e) {
-      // report staging repository failures
-      ErrorDumper.dumpErrors(getLog(), e);
+    final Map<String, String[]> stagingRepositories = getStagingRepositoryIds();
```

Comment:
Instead of using an array of repository ids, the mojo should be able to handle a map of nexurl and repository ids.